compiler/rustc_target/src/spec/mod.rs

```diff
@@ -502,6 +502,26 @@ impl SplitDebuginfo {
             SplitDebuginfo::Unpacked => "unpacked",
         }
     }
+
+    /// Returns `true` if the split debuginfo configuration indicated by `self` is stable on
+    /// `target`.
+    pub fn is_stable_on_platform(&self, target: &TargetOptions) -> bool {
+        // Off, unpacked and packed split debuginfo are stable on OS X-like platforms.
+        if target.is_like_osx {
+            return true;
+        }
+
+        // Packed split debuginfo is the only valid option on Windows platforms.
+        if target.is_like_windows {
+            return match self {
+                SplitDebuginfo::Packed => true,
+                SplitDebuginfo::Off | SplitDebuginfo::Unpacked => false,
+            };
+        }
+
+        // Other platforms use Split DWARF, which is stable.
```

Comment:
I'm slightly unsure about this, but not sure what check would be appropriate.

Reply:
This effectively stabilizes all forms on split debuginfo on platforms like Fuchsia, Solaris, BSD, etc, right? Is that really what we want? The stabilization report only mentions Linux.

Reply:
Yeah, I wasn't sure how best to say "anything that uses DWARF for debuginfo". I've changed this to take a very different approach so that each target now specifies the different values of `SplitDebuginfo` that it supports, that lets us toggle this on in a much more fine-grained way for each target.

Reply:
I suspect it'll take a couple runs through bors to limit the tests to the correct platforms now though..